Source/Analytics/Read/Overview/LastWeeksPerHealthRisk/EventProcessor.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Concepts;
+using Concepts.HealthRisks;
+using Dolittle.Events.Processing;
+using Dolittle.ReadModels;
+using Events.Admin.HealthRisks;
+using Events.Reporting.CaseReports;
+using Read.HealthRisks;
+using HealthRiskName = Read.HealthRisks.HealthRiskName;
+
+namespace Read.Overview.LastWeeksPerHealthRisk
+{
+    public class EventProcessor : ICanProcessEvents
+    {
+        readonly IReadModelRepositoryFor<HealthRiskName> _healthRiskNames;
+        readonly IReadModelRepositoryFor<CaseReportsLastWeeksPerHealthRisk> _caseReportsLastWeeksPerHealthRisk;
+
+        public EventProcessor(
+            IReadModelRepositoryFor<HealthRiskName> healthRiskNames,
+            IReadModelRepositoryFor<CaseReportsLastWeeksPerHealthRisk> caseReportsLastWeeksPerHealthRisk            
+        )
+        {
+            _healthRiskNames = healthRiskNames;
+            _caseReportsLastWeeksPerHealthRisk = caseReportsLastWeeksPerHealthRisk;
+        }
+        
+        [EventProcessor("f2f78ff9-2d3a-a32d-cf9b-37f73451da6c")]
+        public void Process(CaseReportReceived @event)
+        {
+            var healthRiskName = _healthRiskNames.GetById(@event.HealthRiskId)?.Name ?? "Unknown";
+            var recieved = Day.Of(@event.Timestamp);
+            var numberOfCaseReports = @event.NumberOfFemalesUnder5+@event.NumberOfFemalesAged5AndOlder+@event.NumberOfMalesUnder5+@event.NumberOfMalesAged5AndOlder;
+
+            for (var days = 0; days < 7; days++)
+            {
+                CreateOrUpdateCaseReports(recieved+days, @event.HealthRiskId, _ => {
```

Comment:
Is the _ => { ... } thing a Dolittle way of doing things or some fancy new .Net stuff?

Reply:
I feel this is a bit funky.

Reply:
That is an anonymous method. It is not special for Dolittle, and it's not really that new either as it was introduced in C#2.0 almost 15 years ago 😛 

The underscore instead of a "real" name for the method argument might be a bit my coding style. IMO it makes the code more readable when it is clear what the object is, for example have a look at line 100 in the same file: `_caseReportsLastWeeksPerHealthRisk.Query.Where(_ => _.Id >= Day.Today)`. The last part almost reads like you would explain it with words, which I like.

It might look strange how I used it in this particular case, but we could consider renaming some methods. I think the alternative would mean a lot more variables passed around, and harder to read.